Update docs for uenv v10 (#395)
eth-cscs/cscs-docs: docs/software/uenv/using.md

    $ uenv run --view=default prgenv-gnu/24.11:v1 -- ./post-processing-script.sh
    ```

!!! example "using alias to create shortcuts"
    It can be verbose typing the full `uenv run --view=... name/version:tag -- ` in front of each command.

    If you want to use commands from a specific uenv frequently, create an alias:

    ```console
    # create a shortcut alias for running commands in prgenv-gnu
    $ alias prg='uenv run --view=default prgenv-gnu/25.11:v1 --'

    #
    # tools like cmake and gcc can be called: the entire command will run inside
    # the uenv: isolating the call
    #

    $ prg cmake --version
    cmake version 3.31.9
    $ prg which gcc
    /user-environment/env/default/bin/gcc
    ```

    If you have a command that you like to execute from a uenv, it can also be turned into an alias.
    For example, if you want to use `nvim` from the `editors` uenv:

    ```
    $ alias nvim='uenv run --view=ed editors/24.7 -- nvim'

    # now edit a file using neovim from the uenv
    $ nvim main.f90

>     ```

Reviewer: Do we need to add a warning that aliases are ignored inside scripts, unless explicitly enabled?

Author: done!

* `--uenv`:  a comma-separated list of uenv to mount
* `--view`:  a comma-separated list of views to load

> * `--repo`:  customize when repos to search for the requested uenv. See the [`--repo` flag docs][ref-uenv-repo-flag] for more information.

Reviewer: ```suggestion
* `--repo`:  customize the repository order to search for the requested uenv. See the [`--repo` flag docs][ref-uenv-repo-flag] for more information.
```

Author: done.

> * `--uenv-passthrough`: <span class="v-badge">uenv v10</span> configure how sbatch and srun [load or ignore uenv environments][ref-uenv-slurm-passthrough] that are already loaded.

Reviewer: ```suggestion
* `--uenv-passthrough`: <span class="v-badge">uenv v10.0</span> configure how sbatch and srun [load or ignore uenv environments][ref-uenv-slurm-passthrough] that are already loaded.
```

> * `--no-default-view`: <span class="v-badge">uenv v10</span> disable [default views][ref-uenv-views-default].

Reviewer: ```suggestion
* `--no-default-view`: <span class="v-badge">uenv v10.0</span> disable [default views][ref-uenv-views-default].
```

* Note how the second call has access to `mpicc`, provided by `prgenv-gnu`.

[](){#ref-uenv-slurm-envvars}
### Slurm environment variables

The `SBATCH_UENV`, `SBATCH_UENV_VIEW`, and `SBATCH_UENV_REPO` environment variables can be used to pre-configure the uenv for a batch job.
Useful in CI/CD pipelines or wrapper scripts where passing flags to `sbatch` is inconvenient:

The following example is equivalent to adding `--uenv` and `--view` to the sbatch call or inside the sbatch script.

```bash title="setting uenv via environment variables"
export SBATCH_UENV=prgenv-gnu/25.6:v2
export SBATCH_UENV_VIEW=default
sbatch my-job.sh
```

These variables are cleared inside the job, and do not propagate to nested `srun` or `sbatch` calls.

[](){#ref-uenv-slurm-passthrough}
### Passthrough behaviour

> <span class="v-badge">uenv v10</span>

Reviewer: ```suggestion
<span class="v-badge">uenv v10.0</span>
```

Most of the time, the fix for this error is to not start uenv in a running uenv environment.

However, there are workflows and situations where `sbatch` must be called with a uenv mounted.
In these situations, the `--uenv-passthrough` flag can be used to explicitly state how to treat uenv in the calling environment:

!!! example "using `--uenv-passthrough` with `sbatch`"
    ```bash
    #SBATCH --nodes=4
    #SBATCH --ntasks-per-node=4
    #SBATCH --uenv=prgenv-gnu
    #SBATCH --view=default

    # this srun will use prgenv-gnu because srun defaults to --uenv-passthrough=use
    srun ./a.out

    # this srun will not use any uenv
    srun --uenv-passthrough=ignore ./my.exe

    # this srun ignores prgenv-gnu from the calling environment, and uses cp2k instead.
    srun --uenv=cp2k --view=develop ./build

    # start another sbatch job that does not automatically load
    # the prgenv-gnu uenv
    sbatch --uenv-passthrough=ignore ./nested-job

    # start another sbatch job that uses the prgenv-gnu uenv
    sbatch --uenv-passthrough=use ./nested-job
    ```

??? warning "Breaking change: calling `sbatch` from inside a uenv session"

>     In v10, calling `sbatch` or `salloc` from inside an active `uenv start` session is disabled by default and will produce an error:

Reviewer: ```suggestion
    In v10.0, calling `sbatch` or `salloc` from inside an active `uenv start` session is disabled by default and will produce an error:
```

    Some uenv, for example the `prgenv-gnu`, have a view named `default`.

    This view is not loaded by default if no view is specified with the `--view` flag.

    Uenv v10 introduces support for [default views][ref-uenv-views-default].

!!! info "installing Python venv with uenv"
    Python virtual environments can be created on top of a uenv view.
    However, to ensure that the Python interpreter and packages from the uenv view are used, the `PYTHONPATH` and `PYTHONUSERBASE` environment variables must be set correctly, see our guide on [building Python virtual environments with uenv][ref-python-uenv-venv].

[](){#ref-uenv-views-default}
### Default views

> <span class='v-badge'>uenv v10</span>

Reviewer: ```suggestion
<span class='v-badge'>uenv v10.0</span>
```
